### DISCUSSION TITLE
Allow object initializer syntax outside of initialization

### DISCUSSION THREAD

**Discussion by Ekl1p5e** [dotnet/csharplang#4648] [Language Ideas]
I am working with some code that initializes the same object in different places, setting different properties. It would be nice to call a method that sets certain properties and maybe chain them together to eliminate duplicate code. Here is a simple example of how it would work:

class Person
{
 string FirstName {get; set;}
 string LastName {get; set;}
 string MiddleName {get; set;}
 int Age {get; set;}
}

Person GetPerson(string firstName, string lastName)
{
 return new Person
 {
  FirstName = firstName,
  LastName = lastName,
 };
}

Person GetPerson(string firstName, string lastName, int age){
 return GetPerson(firstName, lastName){
  Age = age,
 }
}

I think this would neater in places where multiple properties need to be set on an object as well.

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/803

**Reply by Joe4evr** [reply]
In addition to that one: https://youtu.be/fIXujTOA6BE?t=3023
